src/java.base/share/classes/jdk/internal/vm/annotation/IntrinsicCandidate.java

```diff
@@ -28,94 +28,67 @@
- * intrinsic from object fields must also be checked. As a matter of clarity, it is
- * better to make intrinisics be static methods, to make the dependency on {@code this}
- * clear. Also, it is better to explicitly load all required values from object
- * fields before entering the intrinsic code, and pass those values as explicit arguments.
- * First, this may be necessary for null checks (or other checks). Second, if the
- * intrinsic reloads the values from fields and operates on those without checks,
- * race conditions may be able to introduce unchecked invalid values into the intrinsic.
- * If the intrinsic needs to store a value back to an object field, that value should be
- * returned explicitly from the intrinsic; if there are multiple return values, coders
- * should consider buffering them in an array. Removing field access from intrinsics
- * not only clarifies the interface with between the JVM and JDK; it also helps decouple
- * the HotSpot and JDK implementations, since if JDK code before and after the intrinsic
- * manages all field accesses, then intrinsics can be coded to be agnostic of object
- * layouts.</li>
- *
- * Maintainers of the HotSpot VM must consider the following when modifying
- * intrinsics.
- *
- * <ul>
- * <li>When adding a new intrinsic, make sure that the corresponding method
- * in the Java libraries is annotated with {@code @IntrinsicCandidate}
- * and that all possible call sequences that result in calling the intrinsic contain
- * the checks omitted by the intrinsic (if any).</li>
- * <li>When modifying an existing intrinsic, the Java libraries must be updated
- * to match the semantics of the intrinsic and to execute all checks omitted
- * by the intrinsic (if any).</li>
- * </ul>
- *
- * Persons not directly involved with maintaining the Java libraries or the
- * HotSpot VM can safely ignore the fact that a method is annotated with
- * {@code @IntrinsicCandidate}.
- *
- * The HotSpot VM defines (internally) a list of intrinsics. Not all intrinsic
- * are available on all platforms supported by the HotSpot VM. Furthermore,
- * the availability of an intrinsic on a given platform depends on the
- * configuration of the HotSpot VM (e.g., the set of VM flags enabled).
- * Therefore, annotating a method with {@code @IntrinsicCandidate} does
- * not guarantee that the marked method is intrinsified by the HotSpot VM.
- *
- * If the {@code CheckIntrinsics} VM flag is enabled, the HotSpot VM checks
- * (when loading a class) that (1) all methods of that class that are also on
- * the VM's list of intrinsics are annotated with {@code @IntrinsicCandidate}
- * and that (2) for all methods of that class annotated with
- * {@code @IntrinsicCandidate} there is an intrinsic in the list.
+ * <h2 id="intrinsification">Intrinsification</h2>
+ * The most frequently special treatment is intrinsification, which replaces a
+ * candidate method's body, bytecode or native, with handwritten platform
```

Comment:
Is this sentence missing the word "code" after "native"? Should it have been:

> bytecode or native code, ...

Reply:
This is referring to native method's bodies. I think "bytecode or native" is sufficient to summarize the executable method body in the Java Language/Virtual Machine Specification.